sssd_test_framework/utils/socket.py

```diff
@@ -0,0 +1,245 @@
+        ]:
+            raise ValueError(f"Unknown responder: {responder}")
+
+        self.host = host
+        self.responder = responder
+        self.socket_unit = f"sssd-{responder}.socket"
+        self.service_unit = f"sssd-{responder}.service"
+
+    def enable_socket(self) -> None:
+        """
+        Enable the responder's socket unit.
+        """
+        result = self.host.conn.exec(["systemctl", "enable", self.socket_unit], raise_on_error=False)
+        if result.rc != 0:
+            raise RuntimeError(f"Failed to enable {self.socket_unit}: {result.stderr}")
+
+    def disable_socket(self) -> None:
+        """
+        Disable the responder's socket unit.
+        """
+        result = self.host.conn.exec(["systemctl", "disable", self.socket_unit], raise_on_error=False)
+        if result.rc != 0:
+            raise RuntimeError(f"Failed to disable {self.socket_unit}: {result.stderr}")
+
+    def start_socket(self) -> None:
+        """
+        Start the responder's socket unit.
+        """
+        result = self.host.conn.exec(["systemctl", "start", self.socket_unit], raise_on_error=False)
+        if result.rc != 0:
+            raise RuntimeError(f"Failed to start {self.socket_unit}: {result.stderr}")
+
+    def stop_socket(self) -> None:
+        """
+        Stop the responder's socket unit.
+        """
+        result = self.host.conn.exec(["systemctl", "stop", self.socket_unit], raise_on_error=False)
+        if result.rc != 0:
+            raise RuntimeError(f"Failed to stop {self.socket_unit}: {result.stderr}")
+
+    def is_socket_active(self) -> bool:
+        """
+        Check if the responder's socket unit is active.
+
+        :return: True if active, False otherwise.
+        :rtype: bool
+        """
+        result = self.host.conn.exec(["systemctl", "is-active", self.socket_unit], raise_on_error=False)
+        return result.stdout.strip() == "active"
+
+    def is_service_active(self) -> bool:
+        """
+        Check if the responder's service unit is active.
+
+        :return: True if active, False otherwise.
+        :rtype: bool
+        """
+        result = self.host.conn.exec(["systemctl", "is-active", self.service_unit], raise_on_error=False)
+        return result.stdout.strip() == "active"
+
+    def get_socket_path(self) -> Optional[str]:
+        """
+        Get the UNIX socket path for the responder.
+
+        :return: Socket path if found, None otherwise.
+        :rtype: Optional[str]
+        """
+        result = self.host.conn.exec(
+            ["systemctl", "show", self.socket_unit, "--property=Listen"], raise_on_error=False
+        )
+        if result.rc != 0:
+            raise RuntimeError(f"Failed to get socket path for {self.socket_unit}: {result.stderr}")
+
+        match = re.search(r"Listen=([^\s]+)", result.stdout)
+        return match.group(1) if match else None
+
+    def trigger_socket(self, command: str, args: list[str] = []) -> ProcessResult:
```

Comment:
Imo, if this method is needed at all, it could be intelligent enough to not take any arguments.